actors/common/src/main/java/org/sunbird/learner/actors/LearnerStateActor.java

```diff
@@ -79,6 +71,117 @@ public void onReceive(Request request) throws Exception {
+
+    Map<String, Object> result =
+        ElasticSearchUtil.complexSearch(
+            searchDto,
+            ProjectUtil.EsIndex.sunbird.getIndexName(),
+            ProjectUtil.EsType.usercourses.getTypeName());
+    if (MapUtils.isNotEmpty(result)) {
+      addCourseDetails(request, result);
+    }
+    Response response = new Response();
+    response.put(JsonKey.COURSES, result.get(JsonKey.CONTENT));
+    sender().tell(response, self());
+  }
+
+  private void addCourseDetails(Request request, Map<String, Object> userCoursesResult) {
+    List<Map<String, Object>> batches =
+        (List<Map<String, Object>>) userCoursesResult.get(JsonKey.CONTENT);
+
+    if (CollectionUtils.isNotEmpty(batches)) {
+      return;
+    }
+    String requestBody = prepareCourseSearchRequest(batches);
+    if (StringUtils.isBlank(requestBody)) {
+      return;
+    }
+    ProjectLogger.log(
+        String.format(
+            "requestBody {0} , request param : {1}",
+            requestBody, (String) request.getContext().get(JsonKey.URL_QUERY_STRING)),
+        LoggerEnum.INFO);
+
+    Future<Response> response =
+        ContentSearchUtil.searchContent(
+                (String) request.getContext().get(JsonKey.URL_QUERY_STRING),
+                requestBody,
+                (Map<String, String>) request.getRequest().get(JsonKey.HEADER))
+            .map(mapToCourseByCourseId(), getContext().dispatcher())
+            .map(prepareBatchesResponse(batches), getContext().dispatcher());
+
+    Patterns.pipe(response, getContext().dispatcher());
+  }
+
+  private String prepareCourseSearchRequest(List<Map<String, Object>> batches) {
+    Set<String> courseIds =
+        batches
+            .stream()
+            .map(batch -> (String) batch.get(JsonKey.COURSE_ID))
+            .collect(Collectors.toSet());
+
+    Map<String, Object> filters = new HashMap<String, Object>();
+    filters.put(JsonKey.CONTENT_TYPE, new String[] {JsonKey.COURSE});
+    filters.put(JsonKey.IDENTIFIER, courseIds);
+    Map<String, Map<String, Object>> req = new HashMap<>();
+    req.put(JsonKey.FILTERS, filters);
+    Map<String, Map<String, Map<String, Object>>> reqWrapper = new HashMap<>();
+    reqWrapper.put(JsonKey.REQUEST, req);
+
+    String requestBody = null;
+    try {
+      requestBody = new ObjectMapper().writeValueAsString(reqWrapper);
+    } catch (JsonProcessingException e) {
+      ProjectLogger.log("Exception while processing filters", e);
+    }
+    return requestBody;
+  }
+
+  private Mapper<Map<String, Object>, Map<String, Object>> mapToCourseByCourseId() {
+    return new Mapper<Map<String, Object>, Map<String, Object>>() {
+      @Override
+      public Map<String, Object> apply(Map<String, Object> result) {
+        Map<String, Object> contentsById = new HashMap<>();
+        if (MapUtils.isNotEmpty(result)) {
+
+          ((List<Map<String, Object>>) result.get(JsonKey.CONTENTS))
+              .stream()
+              .forEach(
```

Comment:
An alternate and shorter way to collect map from list of objects

Collectors.toMap(p -> p.getLast(), Function.identity())